src/Internal/ObjectExporter/ClosureExporter.php

```diff
@@ -145,9 +164,52 @@ private function getClosure(array $ast, string $file, int $line, array $path) :
         $closure = $closures[0];
 
         if ($closure->uses) {
-            throw new ExportException("The closure has bound variables through 'use', this is not supported.", $path);
+            $this->closureHandleUses($reflectionFunction, $closure, $path);
         }
 
         return $closure;
     }
+
+    /**
+     * Handle `use` part of closure.
+     *
+     * @param ReflectionFunction $reflectionFunction  Reflection of the closure.
+     * @param Node\Expr\Closure  $closure             Parsed closure.
+     * @param string[]           $path                The path to the closure in the array/object graph.
+     *
+     * @throws ExportException
+     */
+    private function closureHandleUses(
+        ReflectionFunction $reflectionFunction,
+        Node\Expr\Closure $closure,
+        array $path
+    ) : void {
+        if (! $this->exporter->closureUseAsVars) {
+            throw new ExportException("The closure has bound variables through 'use', this is not supported.", $path);
+        }
+
+        $static = $reflectionFunction->getStaticVariables();
```

Comment:
I didn't know that `getStaticVariables()` returned `use` statements, too. Is this documented anywhere? The [manual](https://www.php.net/manual/en/reflectionfunctionabstract.getstaticvariables.php) doesn't say anything about it, so this makes me a bit chilly to rely on it!

This also made me curious: what happens if you have both a `use` statement and a `static` variable with the same name? It looks like `use` wins, so at least we should be safe here:

https://3v4l.org/jbGIf

Reply:
Unfortunately `getStaticVariables()` is one of the many undocumented functions :frowning:

The way `use()` is implemented can only change with a new major version (like PHP 8), as it will break stuff internally (and possibly extensions). There is no way to guarantee varexporter is going to work for PHP 8 with zero modifications. So imho this shouldn't be a deal-breaker.

---

Using `static` on variables that are already defined in the scope doesn't make sense (https://3v4l.org/rXmBE).

Reply:
`getStaticVariables()` returns `op_array.static_variables`. This is intended for variables bound with `static`, but closures also use this array for the `use` variables.

I asked @nikic; This is an implementation detail and might be changed in the future (PHP 8). However, it should always be possible to get these vars using reflection. So if it's changed, a method like `ReflectionFunction::getUseVariables()` would be added.

Reply:
Alright, with good unit tests we should be covered anyway. Maybe it's time to start testing against `nightly` (with `allow_failures`), so that we know right away if this implementation detail changes?